crates/ra_assists/src/assists/add_missing_impl_members.rs

```diff
@@ -227,6 +238,53 @@ fn substitute_type_params<N: AstNode>(
     }
 }
 
+use hir::PathResolution;
+
+// FIXME extract this to a general utility as well
+// FIXME handle value ns?
+// FIXME this doesn't 'commute' with `substitute_type_params`, since type params in newly generated type arg lists don't resolve. Currently we can avoid this problem, but it's worth thinking about a solution
+fn qualify_paths<N: AstNode>(db: &impl HirDatabase, node: hir::InFile<N>, from: hir::Module) -> N {
```

Comment:
Heh, this whole function is a great example of merging ast with hir in a non-trival way....

I wonder if, in the future, we will fall-back to jsut rendering the hir directly to string, disregarding original formatting...